ldms/src/sampler/llnl/lustre_client.c

```diff
@@ -0,0 +1,360 @@
+                }
+                if (llite == NULL)
+                        continue;
+                rbt_ins(&new_llite_tree, &llite->llite_tree_node);
+        }
+        closedir(dir);
+
+        /* destroy any llites remaining in the global llite_tree since we
+           did not see their associated directories this time around */
+        /* llites_destroy();*/
+        /* NOTE: ldms_set_delete() doesn't really work until some time post v4.2.
+           So instead of llites_destroy(), we'll just keep old data around forever
+           in this version of the code.  When ldms_set delete() works we can
+           delete the following while() loop and uncomment the llites_destroy()
+           call above. */
+        while (!rbt_empty(&llite_tree)) {
+                struct rbn *rbn;
+                rbn = rbt_min(&llite_tree);
+                rbt_del(&llite_tree, rbn);
+                rbt_ins(&new_llite_tree, rbn);
+        }
+
+        /* copy the new_llite_tree into place over the global llite_tree */
+        memcpy(&llite_tree, &new_llite_tree, sizeof(struct rbt));
+
+        return;
+}
+
+static void llites_sample()
+{
+        struct rbn *rbn;
+
+        /* walk tree of known LLITEs */
+        RBT_FOREACH(rbn, &llite_tree) {
+                struct llite_data *llite;
+                llite = container_of(rbn, struct llite_data, llite_tree_node);
+                llite_general_sample(llite->name, llite->stats_path,
+                                   llite->general_metric_set);
+        }
+}
+
+static int sample(struct ldmsd_sampler *self)
+{
+        log_fn(LDMSD_LDEBUG, SAMP" sample() called\n");
+        if (llite_general_schema_is_initialized() < 0) {
+                if (llite_general_schema_init() < 0) {
+                        log_fn(LDMSD_LERROR, SAMP" general schema create failed\n");
+                        return ENOMEM;
+                }
+        }
+
+        if ((time(NULL)-last_refresh) >= refresh_interval) {
+                last_refresh = time(NULL);
+                llites_refresh();
+        }
+
+        llites_sample();
+
+        return 0;
+}
+
+static void term(struct ldmsd_plugin *self)
+{
+        log_fn(LDMSD_LDEBUG, SAMP" term() called\n");
+        llites_destroy();
+        llite_general_schema_fini();
+}
+
+static ldms_set_t get_set(struct ldmsd_sampler *self)
+{
+	return NULL;
+}
+
+static const char *usage(struct ldmsd_plugin *self)
+{
+        log_fn(LDMSD_LDEBUG, SAMP" usage() called\n");
+	return  "config name=" SAMP " [producer=<prod_name>] [instance=<inst_name>] [component_id=<compid>] [schema=<sname>] [rescan_sec=<rsec>] [with_jobid=<jid>]\n"
+		"    <prod_name>  The producer name (default: the hostname)\n"
+		"    <inst_name>  The instance name (ignored, dynamically generated)\n"
+		"    <compid>     Optional unique number identifier (default: zero)\n"
+		"    <sname>      Optional schema name (ignored, static schema name)\n"
```

Comment:
must not be ignored. allowing user tuning schema name would help address Tom's comment if the schema evolves or has optional parts in the future.

Reply:
This would seem to be unrelated to Tom's comment.  Tom's comment was about when the plugin author changes the schema.  When that happens, the plugin author is responsible for changing the schema name.  Understood.  Pushing that responsibility onto users to change the schema name when the plugin author makes a schema change seems undesirable.  How many users can really be expected to figure out which schema changes have happened in a a schema that hasn't changed names, and then use a configuration option to change the schema name themselves?  Seems like an odd design choice.

But this one is an easy change, so even if I don't see what use it has, I'll add this tomorrow.